packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -267,39 +261,46 @@ export async function sideloadAddIn(manifestPath: string, app?: OfficeApp, canPr
     isDesktop = false;
   }
 
-  if (app) {
-    if (appsInManifest.indexOf(app) < 0) {
-      throw new Error(`The Office Add-in manifest does not support ${getOfficeAppName(app)}.`);
-    }
-  } else {
-    switch (appsInManifest.length) {
-      case 0:
-        throw new Error("The manifest does not support any Office apps.");
-      case 1:
-        app = appsInManifest[0];
-        break;
-      default:
-        if (canPrompt) {
-          app = await chooseOfficeApp(appsInManifest);
-        } else {
-          throw new Error("Please specify the Office app.");
-        }
-        break;
+  if (platform && platform == AppType.Web && document === undefined) {
```

Comment:
I think this would be better in the else clause below just before the generateSideloadUrl() call.